posts/MichelaRocchettiBlogPost/Michela_JSoC.qmd

```diff
@@ -0,0 +1,111 @@
+a structure for the webpage explaining the IPUMS.jl package. In this prespective, some pages were added:
+  1. Home: to explain the main purpose of this work and to give general informations.
+  2. Workflows: to explain the work process behind.
+  3. How to: to give general information on how to make things work.
+  4. Tutorials: some tutorials on what is possible to do with IPUMS.jl and how to make it work 
+  5. Examples: some activity examples.
+  6. Mission: to better explain why IPUMS.jl package is useful for the community and its final objective and possible implications
+  7. References: references used to write the pages down.
+
+This first task requires some time, in particular to set up github  and cloning the repository locally in a correct manner.
+
+The second task consisted in updating the documentation across IPUMS.jl.To do that, function within the model folder in the IPUMS.jl folder were modified. The main objective of this task was to provide
+a description of the function and its attributes and an example of possible implementation and outcome. The documentation to be updated was of various type of function:
+  1. DataExtract
+  2. Dataset
+  3. DataTable
+  4. TimeSeriesTable
+  5. Error
+  6. Shapefile.
+  Each of those macrocategories (from 1 to 4) contains a set of function each, signaling the  different expected output and specific purpose.
+  Information about what each function does and about the meaning of each specific input variable, were found on the IPUMS website and references were given in writing down
+  the documentation. 
+  After writing down the description of function and inputs, examples were then formulated starting from IPUMS website: signing up to ipums, an API key 
+  is given and this is used,among the other things, to run pre-written code on the website. This code is concenring examples of those functions, and those examples
+  were re-adapted by changing some input values and fitting them to work in Julia framework.The latter task was done simply rewriting some structures, such as dictionary,maps or lists, in
+  Julia language. 
+
+  A last task consisted in testing functions in the 'api_IPUMSAPI.jl' file. In this file, the function to be tested and other functions are defined and the main ones are extracted to be
+  available in the whole framework. The functions to be tested were the following:
+    1. metadata_nhgis_data_tables_get
+    2. metadata_nhgis_datasets_dataset_data_tables_data_table_get
+    3. metadata_nhgis_datasets_dataset_get
+    4. metadata_nhgis_datasets_get
+    All those functions defined in the .jl file are coming from the `api_metadata.R` file of the ipumsr package. 
+    Before working on the Julia files, a work of testing and undestanding of the original R function was done using R studio. 
```

Comment:
Oh could you include maybe an example of calling a function here?

Reply:
Could you also include a screen capture of you working with R Studio to validate our implementations?